### PR TITLE
Sub-networks implementation

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the new behavior (if this is a feature change)?**
A new merging with sub network feature has been added in powsybl core 6.0.0. This PR indent to interface this new feature to Python.

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: java/src/main/java/com/powsybl/python/commons/Util.java [hunk 1]
```diff
@@ -309,6 +309,8 @@ public static DataframeElementType convert(PyPowsyblApiHeader.ElementType type)
                 return DataframeElementType.BRANCH;
             case IDENTIFIABLE:
                 return DataframeElementType.IDENTIFIABLE;
+            case SUB_NETWORK:
+                return DataframeElementType.SUB_NETWORK;
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/python/commons/Util.java]: no need in the inverse converter ?

Comment (geofjamg) [reply java/src/main/java/com/powsybl/python/commons/Util.java]: Fixed

Comment (EtienneLt) [reply java/src/main/java/com/powsybl/python/commons/Util.java:315]: well I think this dataframe is not updated so yes

File: pypowsybl/network/impl/network.py [hunk 2]
```diff
@@ -306,6 +306,39 @@ def get_elements(self, element_type: ElementType, all_attributes: bool = False,
             result = result[attributes]
         return result
 
+    def get_sub_networks(self, all_attributes: bool = False, attributes: List[str] = None,
+                         **kwargs: ArrayLike) -> DataFrame:
+        """
+        Get a dataframe of sub networks
```
Comment (EtienneLt) [inline pypowsybl/network/impl/network.py:312]: the dataframe just lists the names of the subnetwork ?

File: pypowsybl/network/impl/network.py [hunk 3]
```diff
@@ -306,6 +306,39 @@ def get_elements(self, element_type: ElementType, all_attributes: bool = False,
...
+
+        Args:
+            all_attributes: flag for including all attributes in the dataframe, default is false
+            attributes: attributes to include in the dataframe. The 2 parameters are mutually exclusive.
+                        If no parameter is specified, the dataframe will include the default attributes.
+
+        Returns:
+            A dataframe of sub networks.
+        """
```
Comment (EtienneLt) [inline pypowsybl/network/impl/network.py:321]: I'm not sure it is usefull but maybe a description of the column could be added 
for example 
 
Notes:
            The resulting dataframe, depending on the parameters, will include the following columns:

            - **id**: the subnetwork id

File: pypowsybl/network/impl/network.py [hunk 4]
```diff
@@ -306,6 +306,39 @@ def get_elements(self, element_type: ElementType, all_attributes: bool = False,
...
+            sub_network_id: the id of the sub network
+
+        Returns:
+            The sub network.
+        """
+        return Network(_pp.get_sub_network(self._handle, sub_network_id))
+
+    def detach(self) -> None:
+        """
+        Detach a sub network from its parent network.
```
Comment (EtienneLt) [inline pypowsybl/network/impl/network.py:338]: what does it mean exactly to detach a subnework ?

File: pypowsybl/network/impl/network.py (hunk 2 above)
Comment (geofjamg) [reply pypowsybl/network/impl/network.py:312]: Yes, there is no column (maybe one day), just the index is the ID of the sub networks

File: pypowsybl/network/impl/network.py (hunk 3 above)
Comment (geofjamg) [reply pypowsybl/network/impl/network.py:321]: there is no column just an index

File: pypowsybl/network/impl/network.py (hunk 4 above)
Comment (geofjamg) [reply pypowsybl/network/impl/network.py:338]: It means it is no more merged with other networks

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:338]: is it clear for a user ?

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:338]: maybe it is in the documentation ?

Comment (geofjamg) [reply pypowsybl/network/impl/network.py:338]: There is a little bit more infos in the documentation. If you think more details are needed here, do not hesitate to update it as you would like to have.

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**